pkg/provider/acme/local_store.go

```diff
@@ -31,10 +31,16 @@ func NewLocalStore(filename string) *LocalStore {
 
 func (s *LocalStore) save(resolverName string, storedData *StoredData) {
 	s.lock.Lock()
-	defer s.lock.Unlock()
-
 	s.storedData[resolverName] = storedData
-	s.saveDataChan <- s.storedData
+	s.lock.Unlock()
+
+	s.lock.RLock()
+	storedDataCopy := s.unSafeCopyOfStoredData(s.storedData)
+	s.lock.RUnlock()
```

Comment:
Here it seems unnecessary to differentiate the lock usage, so reintroduce the `defer the s.lock.Unlock` seems better to me, in place of this code.